src/database/abstract.module.ts

```diff
@@ -0,0 +1,68 @@
+import { Module, DynamicModule } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { MongooseModule } from '@nestjs/mongoose';
+import { MongoDbService } from './mongo-db.service';
+import { AbstractDbService } from './abstract-db.service';
+import {
+  User,
+  UserSchema,
+  UserDocument,
+} from 'src/api/users/schemas/user.schema';
+import { Model } from 'mongoose';
+
+@Module({})
+export class DbModule {
+  static forRoot(): DynamicModule {
+    return {
+      module: DbModule,
+      imports: [
+        MongooseModule.forRootAsync({
+          useFactory: (configService: ConfigService) => ({
+            uri: configService.get<string>('MONGO_URI'),
+          }),
+          inject: [ConfigService],
+        }),
+        MongooseModule.forFeature([{ name: User.name, schema: UserSchema }]),
+      ],
+      providers: [
+        {
+          provide: AbstractDbService,
+          useFactory: async (
+            configService: ConfigService,
+            userModel: Model<UserDocument>,
+          ) => {
+            const dbType = configService.get<string>('DB_TYPE');
+            if (dbType === 'mongo') {
+              return new MongoDbService(userModel);
+            } else {
+              throw new Error('Непідтримуваний тип бази даних.');
+            }
+          },
+          inject: [ConfigService, 'UserModel'],
+        },
+        MongoDbService,
+      ],
+      exports: [AbstractDbService, MongooseModule],
+    };
+  }
+}
```

Comment:
IN this abstraction you just set DB module to use mongodb, in this case it's not an abstraction anymore)
Please review again my example
There I set dynamically what DB layer to use 
And I did this configuration in **_ResidentsService_**